### PR TITLE
Feature: adding notional v3 yield

### PR DESCRIPTION
**Pull request by jeffywu**

### CODE REVIEW

**Comment by llamatester** [comment]
The notional-v2 adapter exports pools: 
        
 
 Test Suites: 1 passed, 1 total
    Tests:       92 passed, 92 total
    Snapshots:   0 total
    Time:        0.284 s
    Ran all test suites.
    
    Nb of pools: 14
     
    
    Sample pools:
    ┌─────────┬──────────────────────────────────────┬────────────┬───────────────┬─────────┬──────────────────┬───────────────────────────┬───────────────────┬──────────────────────────────────────────────────┬──────────────────────────────────────────────────┬───────────────────────┬───────────────────────────────────────────────────────────────┐
    │ (index) │                 pool                 │   chain    │    project    │ symbol  │      tvlUsd      │          apyBase          │     apyReward     │                   rewardTokens                   │                 underlyingTokens                 │       poolMeta        │                              url                              │
    ├─────────┼──────────────────────────────────────┼────────────┼───────────────┼─────────┼──────────────────┼───────────────────────────┼───────────────────┼──────────────────────────────────────────────────┼──────────────────────────────────────────────────┼───────────────────────┼───────────────────────────────────────────────────────────────┤
    │    0    │           'nUSDC-notional'           │ 'Ethereum' │ 'notional-v2' │ 'nUSDC' │ 8479766.92656181 │         0.0885561         │     8.635071      │ [ '0xCFEAead4947f0705A14ec42aC3D44129E1Ef3eD5' ] │ [ '0xa0b86991c6218b36c1d19d4a2e9eb0ce3606eb48' ] │         null          │           'https://www.notional.finance/lend/nUSDC'           │
    │    1    │           'nETH-notional'            │ 'Ethereum' │ 'notional-v2' │ 'nETH'  │ 6184171.51284746 │         0.2763298         │     4.305619      │ [ '0xCFEAead4947f0705A14ec42aC3D44129E1Ef3eD5' ] │ [ '0xaac5145f5286a3c6a06256fdfbf5b499aa965c9c' ] │         null          │           'https://www.notional.finance/lend/nETH'            │
    │    2    │ 'fUSDC-Maturing-2023-12-19-notional' │ 'Ethereum' │ 'notional-v2' │ 'fUSDC' │ 4748860.75281776 │         5.0754678         │       null        │                       null                       │ [ '0xa0b86991c6218b36c1d19d4a2e9eb0ce3606eb48' ] │ 'Maturing 2023-12-19' │ 'https://www.notional.finance/lend/fUSDC,Maturing,2023-12-19' │
    │    3    │           'nDAI-notional'            │ 'Ethereum' │ 'notional-v2' │ 'nDAI'  │ 3904804.89119496 │        -0.0420718         │ 8.523693999999999 │ [ '0xCFEAead4947f0705A14ec42aC3D44129E1Ef3eD5' ] │ [ '0x6b175474e89094c44da98b954eedeac495271d0f' ] │         null          │           'https://www.notional.finance/lend/nDAI'            │
    │    4    │ 'fETH-Maturing-2023-12-19-notional'  │ 'Ethereum' │ 'notional-v2' │ 'fETH'  │ 3163922.96042917 │         4.0644614         │       null        │                       null                       │ [ '0xaac5145f5286a3c6a06256fdfbf5b499aa965c9c' ] │ 'Maturing 2023-12-19' │ 'https://www.notional.finance/lend/fETH,Maturing,2023-12-19'  │
    │    5    │ 'fUSDC-Maturing-2024-03-18-notional' │ 'Ethereum' │ 'notional-v2' │ 'fUSDC' │ 2603527.25012401 │         6.4784673         │       null        │                       null                       │ [ '0xa0b86991c6218b36c1d19d4a2e9eb0ce3606eb48' ] │ 'Maturing 2024-03-18' │ 'https://www.notional.finance/lend/fUSDC,Maturing,2024-03-18' │
    │    6    │ 'fETH-Maturing-2024-03-18-notional'  │ 'Ethereum' │ 'notional-v2' │ 'fETH'  │ 2599372.12766395 │         2.9586793         │       null        │                       null                       │ [ '0xaac5145f5286a3c6a06256fdfbf5b499aa965c9c' ] │ 'Maturing 2024-03-18' │ 'https://www.notional.finance/lend/fETH,Maturing,2024-03-18'  │
    │    7    │ 'fDAI-Maturing-2023-12-19-notional'  │ 'Ethereum' │ 'notional-v2' │ 'fDAI'  │ 2079829.7500822  │         4.3772097         │       null        │                       null                       │ [ '0x6b175474e89094c44da98b954eedeac495271d0f' ] │ 'Maturing 2023-12-19' │ 'https://www.notional.finance/lend/fDAI,Maturing,2023-12-19'  │
    │    8    │           'nWBTC-notional'           │ 'Ethereum' │ 'notional-v2' │ 'nWBTC' │ 1917746.93141285 │ -0.0000038000000000000005 │         0         │ [ '0xCFEAead4947f0705A14ec42aC3D44129E1Ef3eD5' ] │ [ '0x2260fac5e5542a773aa44fbcfedf7c193bc2c599' ] │         null          │           'https://www.notional.finance/lend/nWBTC'           │
    │    9    │ 'fDAI-Maturing-2024-03-18-notional'  │ 'Ethereum' │ 'notional-v2' │ 'fDAI'  │ 1497410.70771163 │     5.204586600000001     │       null        │                       null                       │ [ '0x6b175474e89094c44da98b954eedeac495271d0f' ] │ 'Maturing 2024-03-18' │ 'https://www.notional.finance/lend/fDAI,Maturing,2024-03-18'  │
    └─────────┴──────────────────────────────────────┴────────────┴───────────────┴─────────┴──────────────────┴───────────────────────────┴───────────────────┴──────────────────────────────────────────────────┴──────────────────────────────────────────────────┴───────────────────────┴───────────────────────────────────────────────────────────────┘

File: src/adaptors/notional-v3/index.js [hunk 1]
```diff
@@ -0,0 +1,197 @@
+const { request, gql } = require('graphql-request');
+const utils = require('../utils');
+const superagent = require('superagent');
+const { default: BigNumber } = require('bignumber.js');
+
+const API = (chain) => `https://data-dev.notional.finance/${chain}/yields`
+const NOTE_Mainnet = '0xCFEAead4947f0705A14ec42aC3D44129E1Ef3eD5'
+
+const SUBGRAPHS = {
+  arbitrum: 'https://api.studio.thegraph.com/query/33671/notional-finance-v3-arbitrum/version/latest'
```
Comment (jeffywu) [inline src/adaptors/notional-v3/index.js]: FYI: this is a development URL in studio, are you guys able to support queries against the decentralized subgraph network or do you need to use the hosted network only?

**Comment by llamatester** [comment]
The notional-v3 adapter exports pools: 
        
 
 Test Suites: 1 passed, 1 total
    Tests:       247 passed, 247 total
    Snapshots:   0 total
    Time:        0.33 s
    Ran all test suites.
    
    Nb of pools: 39
     
    
    Sample pools:
    ┌─────────┬───────────────────────────────────────────────────────────────────────────────┬────────────┬───────────────┬────────────────────────────────┬──────────────────────────────────────────────────┬──────────────────────────────────────────────────┬───────────────────┬───────────────────────────────────────────────────────────────┬────────────────────┬─────────────┬──────────────────────┬───────────────┬────────────────────┬────────────────────┐
    │ (index) │                                     pool                                      │   chain    │    project    │             symbol             │                   rewardTokens                   │                 underlyingTokens                 │     poolMeta      │                              url                              │       tvlUsd       │   apyBase   │      apyReward       │ apyBaseBorrow │   totalSupplyUsd   │   totalBorrowUsd   │
    ├─────────┼───────────────────────────────────────────────────────────────────────────────┼────────────┼───────────────┼────────────────────────────────┼──────────────────────────────────────────────────┼──────────────────────────────────────────────────┼───────────────────┼───────────────────────────────────────────────────────────────┼────────────────────┼─────────────┼──────────────────────┼───────────────┼────────────────────┼────────────────────┤
    │    0    │             '0x0f13fb925edc3e1fe947209010d9c0e072986adc-arbitrum'             │ 'arbitrum' │ 'notional-v3' │            'nUSDC'             │ [ '0x019be259bc299f3f653688c7655c87f998bc7bc1' ] │ [ '0xaf88d065e77c8cc2239327c5edb3a432268e5831' ] │ 'Liquidity Token' │  'https://arbitrum.notional.finance/liquidity-variable/USDC'  │ 1085677.503452883  │ 0.049962592 │ 0.059757155923014955 │               │                    │                    │
    │    1    │             '0x18b0fc5a233acf1586da7c199ca9e3f486305a29-arbitrum'             │ 'arbitrum' │ 'notional-v3' │             'nETH'             │ [ '0x019be259bc299f3f653688c7655c87f998bc7bc1' ] │ [ '0x0000000000000000000000000000000000000000' ] │ 'Liquidity Token' │  'https://arbitrum.notional.finance/liquidity-variable/ETH'   │ 891528.5944066945  │ 0.025346682 │  0.0727705206067761  │               │                    │                    │
    │    2    │ '0x0000000000000000000000000000000000000000000000000003006580dd0001-arbitrum' │ 'arbitrum' │ 'notional-v3' │ 'fUSDC Maturing On 2023-12-19' │                                                  │ [ '0xaf88d065e77c8cc2239327c5edb3a432268e5831' ] │   'Fixed Lend'    │      'https://arbitrum.notional.finance/lend-fixed/USDC'      │ 578823.7508286423  │ 0.050086407 │                      │  0.050086407  │ 1034279.4417246009 │ 1037406.7332728102 │
    │    3    │ '0x0000000000000000000000000000000000000000000000000001006580dd0001-arbitrum' │ 'arbitrum' │ 'notional-v3' │ 'fETH Maturing On 2023-12-19'  │                                                  │ [ '0x0000000000000000000000000000000000000000' ] │   'Fixed Lend'    │      'https://arbitrum.notional.finance/lend-fixed/ETH'       │  468842.893985895  │ 0.028273319 │                      │  0.028273319  │ 168920.48028241956 │ 179398.4057593593  │
    │    4    │             '0x06d45ef1f8b3c37b0de66f156b11f10b4837619a-arbitrum'             │ 'arbitrum' │ 'notional-v3' │           'nwstETH'            │ [ '0x019be259bc299f3f653688c7655c87f998bc7bc1' ] │ [ '0x5979d7b546e38e414f7e9822514be443a4800529' ] │ 'Liquidity Token' │ 'https://arbitrum.notional.finance/liquidity-variable/wstETH' │ 463913.7077954912  │ 0.011481731 │ 0.05593885143342096  │               │                    │                    │
    │    5    │             '0xabc07bf91469c5450d6941dd0770e6e6761b90d6-arbitrum'             │ 'arbitrum' │ 'notional-v3' │             'pETH'             │                                                  │ [ '0x0000000000000000000000000000000000000000' ] │  'Variable Lend'  │     'https://arbitrum.notional.finance/lend-variable/ETH'     │ 437141.0528272736  │  0.0176444  │                      │  0.041901412  │  883669.967713143  │ 446528.9148858694  │
    │    6    │ '0x00000000000000000000000000000000000000000000000000030065f7840001-arbitrum' │ 'arbitrum' │ 'notional-v3' │ 'fUSDC Maturing On 2024-03-18' │                                                  │ [ '0xaf88d065e77c8cc2239327c5edb3a432268e5831' ] │   'Fixed Lend'    │      'https://arbitrum.notional.finance/lend-fixed/USDC'      │ 431565.57989026944 │ 0.059104968 │                      │  0.059104968  │ 1615919.5361056393 │ 1616350.9397572041 │
    │    7    │ '0x00000000000000000000000000000000000000000000000000010065f7840001-arbitrum' │ 'arbitrum' │ 'notional-v3' │ 'fETH Maturing On 2024-03-18'  │                                                  │ [ '0x0000000000000000000000000000000000000000' ] │   'Fixed Lend'    │      'https://arbitrum.notional.finance/lend-fixed/ETH'       │  413461.107431853  │ 0.041611412 │                      │  0.041611412  │ 338212.6551607047  │ 350486.8344802226  │
    │    8    │             '0x6f6603f12af215bdba1f55f643e098530dd45b8f-arbitrum'             │ 'arbitrum' │ 'notional-v3' │            'nrETH'             │ [ '0x019be259bc299f3f653688c7655c87f998bc7bc1' ] │ [ '0xec70dcb4a1efa46b8f2d97c310c9c4790ba5ffa8' ] │ 'Liquidity Token' │  'https://arbitrum.notional.finance/liquidity-variable/rETH'  │ 325523.04853565997 │ 0.022629985 │ 0.07972031504182289  │               │                    │                    │
    │    9    │             '0x6f28cafe12bd97e474a52bcbfea6f2c18ae0f53d-arbitrum'             │ 'arbitrum' │ 'notional-v3' │            'pUSDC'             │                                                  │ [ '0xaf88d065e77c8cc2239327c5edb3a432268e5831' ] │  'Variable Lend'  │    'https://arbitrum.notional.finance/lend-variable/USDC'     │ 305940.9326664032  │ 0.032378226 │                      │  0.055715935  │ 1010895.5546451912 │  704954.621978788  │
    └─────────┴───────────────────────────────────────────────────────────────────────────────┴────────────┴───────────────┴────────────────────────────────┴──────────────────────────────────────────────────┴──────────────────────────────────────────────────┴───────────────────┴───────────────────────────────────────────────────────────────┴────────────────────┴─────────────┴──────────────────────┴───────────────┴────────────────────┴────────────────────┘
    This adapter contains some pools with <10k TVL, these pools won't be shown in DefiLlama

**Comment by slasher125** [comment]
hi, ty for adding
- apy needs to be scaled by *100 (in pct)
- pls move things like `Maturing On 2023-12-19` into poolMeta

**Comment by llamatester** [comment]
The notional-v3 adapter exports pools: 
        
 
 Test Suites: 1 passed, 1 total
    Tests:       247 passed, 247 total
    Snapshots:   0 total
    Time:        0.321 s
    Ran all test suites.
    
    Nb of pools: 39
     
    
    Sample pools:
    ┌─────────┬───────────────────────────────────────────────────────────────────────────────┬────────────┬───────────────┬───────────┬──────────────────────────────────────────────────┬──────────────────────────────────────────────────┬─────────────────────────────────────┬───────────────────────────────────────────────────────────────┬────────────────────┬───────────┬───────────────────┬───────────────┬────────────────────┬────────────────────┐
    │ (index) │                                     pool                                      │   chain    │    project    │  symbol   │                   rewardTokens                   │                 underlyingTokens                 │              poolMeta               │                              url                              │       tvlUsd       │  apyBase  │     apyReward     │ apyBaseBorrow │   totalSupplyUsd   │   totalBorrowUsd   │
    ├─────────┼───────────────────────────────────────────────────────────────────────────────┼────────────┼───────────────┼───────────┼──────────────────────────────────────────────────┼──────────────────────────────────────────────────┼─────────────────────────────────────┼───────────────────────────────────────────────────────────────┼────────────────────┼───────────┼───────────────────┼───────────────┼────────────────────┼────────────────────┤
    │    0    │             '0x0f13fb925edc3e1fe947209010d9c0e072986adc-arbitrum'             │ 'arbitrum' │ 'notional-v3' │  'nUSDC'  │ [ '0x019be259bc299f3f653688c7655c87f998bc7bc1' ] │ [ '0xaf88d065e77c8cc2239327c5edb3a432268e5831' ] │          'Liquidity Token'          │  'https://arbitrum.notional.finance/liquidity-variable/USDC'  │ 1085923.871813368  │ 5.0012215 │ 5.943694725449379 │               │                    │                    │
    │    1    │             '0x18b0fc5a233acf1586da7c199ca9e3f486305a29-arbitrum'             │ 'arbitrum' │ 'notional-v3' │  'nETH'   │ [ '0x019be259bc299f3f653688c7655c87f998bc7bc1' ] │ [ '0x0000000000000000000000000000000000000000' ] │          'Liquidity Token'          │  'https://arbitrum.notional.finance/liquidity-variable/ETH'   │ 921864.0618243053  │ 2.5135724 │ 7.001466122345509 │               │                    │                    │
    │    2    │ '0x0000000000000000000000000000000000000000000000000003006580dd0001-arbitrum' │ 'arbitrum' │ 'notional-v3' │  'fUSDC'  │                                                  │ [ '0xaf88d065e77c8cc2239327c5edb3a432268e5831' ] │ 'Fixed Lend Maturing On 2023-12-19' │      'https://arbitrum.notional.finance/lend-fixed/USDC'      │ 578859.1232181088  │ 5.0085864 │                   │   5.0085864   │ 1034661.3682491584 │ 1037789.1976683211 │
    │    3    │ '0x0000000000000000000000000000000000000000000000000001006580dd0001-arbitrum' │ 'arbitrum' │ 'notional-v3' │  'fETH'   │                                                  │ [ '0x0000000000000000000000000000000000000000' ] │ 'Fixed Lend Maturing On 2023-12-19' │      'https://arbitrum.notional.finance/lend-fixed/ETH'       │ 491139.2301259158  │ 2.7575273 │                   │   2.7575273   │ 168339.72820495936 │ 185517.88642551965 │
    │    4    │             '0x06d45ef1f8b3c37b0de66f156b11f10b4837619a-arbitrum'             │ 'arbitrum' │ 'notional-v3' │ 'nwstETH' │ [ '0x019be259bc299f3f653688c7655c87f998bc7bc1' ] │ [ '0x5979d7b546e38e414f7e9822514be443a4800529' ] │          'Liquidity Token'          │ 'https://arbitrum.notional.finance/liquidity-variable/wstETH' │ 475679.3757447207  │ 1.1504792 │ 5.427521417557048 │               │                    │                    │
    │    5    │             '0xabc07bf91469c5450d6941dd0770e6e6761b90d6-arbitrum'             │ 'arbitrum' │ 'notional-v3' │  'pETH'   │                                                  │ [ '0x0000000000000000000000000000000000000000' ] │           'Variable Lend'           │     'https://arbitrum.notional.finance/lend-variable/ETH'     │ 458336.8988263926  │ 1.742749  │                   │   4.1671506   │  920084.423749956  │ 461747.5249235634  │
    │    6    │ '0x00000000000000000000000000000000000000000000000000030065f7840001-arbitrum' │ 'arbitrum' │ 'notional-v3' │  'fUSDC'  │                                                  │ [ '0xaf88d065e77c8cc2239327c5edb3a432268e5831' ] │ 'Fixed Lend Maturing On 2024-03-18' │      'https://arbitrum.notional.finance/lend-fixed/USDC'      │ 431667.16913215007 │ 5.9104968 │                   │   5.9104968   │ 1616427.4815282903 │ 1616858.9708493466 │
    │    7    │ '0x00000000000000000000000000000000000000000000000000010065f7840001-arbitrum' │ 'arbitrum' │ 'notional-v3' │  'fETH'   │                                                  │ [ '0x0000000000000000000000000000000000000000' ] │ 'Fixed Lend Maturing On 2024-03-18' │      'https://arbitrum.notional.finance/lend-fixed/ETH'       │ 427529.9649355254  │ 4.1611412 │                   │   4.1611412   │  349744.80159526   │ 362437.4975820783  │
    │    8    │             '0x6f6603f12af215bdba1f55f643e098530dd45b8f-arbitrum'             │ 'arbitrum' │ 'notional-v3' │  'nrETH'  │ [ '0x019be259bc299f3f653688c7655c87f998bc7bc1' ] │ [ '0xec70dcb4a1efa46b8f2d97c310c9c4790ba5ffa8' ] │          'Liquidity Token'          │  'https://arbitrum.notional.finance/liquidity-variable/rETH'  │ 335327.2111814313  │ 2.2120421 │ 7.699226046035522 │               │                    │                    │
    │    9    │             '0x6f28cafe12bd97e474a52bcbfea6f2c18ae0f53d-arbitrum'             │ 'arbitrum' │ 'notional-v3' │  'pUSDC'  │                                                  │ [ '0xaf88d065e77c8cc2239327c5edb3a432268e5831' ] │           'Variable Lend'           │    'https://arbitrum.notional.finance/lend-variable/USDC'     │ 305964.3694106851  │ 3.2380027 │                   │   5.5717183   │ 1011050.4244032349 │ 705086.0549925497  │
    └─────────┴───────────────────────────────────────────────────────────────────────────────┴────────────┴───────────────┴───────────┴──────────────────────────────────────────────────┴──────────────────────────────────────────────────┴─────────────────────────────────────┴───────────────────────────────────────────────────────────────┴────────────────────┴───────────┴───────────────────┴───────────────┴────────────────────┴────────────────────┘
    This adapter contains some pools with <10k TVL, these pools won't be shown in DefiLlama

**Pull request merged**